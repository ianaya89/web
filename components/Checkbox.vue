<template>
  <v-checkbox
    v-model="valueData"
    class="checkbox"
    :class="[requiredClass]"
    v-bind="$props"
    :value="checkboxValue"
    :error-messages="translatableErrorMessages"
    @change="valueChanged"
  >
    <template v-slot:label>
      <slot name="label" />
    </template>
  </v-checkbox>
</template>

<script>
import BaseInput from '@/components/BaseInput'

export default BaseInput.extend({
  name: 'Checkbox',

  props: {
    color: {
      type: String,
      default: 'red'
    },
    checkboxValue: {
      type: [String, Number],
      required: true
    },
    indeterminate: {
      type: Boolean,
      default: false
    },
    value: {
      type: [String, Number, Array, Boolean],
      default: null
    },
    errorMessages: {
      type: Array,
      default() {
        return []
      }
    }
  },

  data: () => ({
    valueData: null
  }),

  watch: {
    value(newVal) {
      this.valueData = newVal
    }
  },
  computed: {
    translatableErrorMessages() {
      return this.errorMessages.map(message => this.$t(message))
    },

    requiredClass() {
      return this.required ? 'required-sign' : ''
    }
  },
  mounted() {
    this.valueData = this.value
  },

  methods: {
    valueChanged(value) {
      this.$emit('input', value)
    }
  }
})
</script>

<style lang="postcss">
.checkbox {
  &.v-input--has-state {
    .v-input__control {
      .v-input__slot {
        &::before {
          @apply .border-red;
        }
      }
    }

    .v-messages__message {
      @apply .text-red .leading-tight;
    }
  }

  .v-input__control .v-input__slot {
    @apply .mb-0;

    .red--text {
      @apply .text-red;
    }

    .v-messages {
      @apply .pt-1 .leading-tight;
    }

    .theme--light.fa-minus-square.v-icon {
      @apply .text-red;
    }

    .required-sign {
      label::after {
        @apply .text-red;
        content: '*';
      }
    }
  }
}
</style>
