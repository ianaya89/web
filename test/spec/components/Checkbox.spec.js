import { mount, shallowMount } from '@vue/test-utils'
import Checkbox from '@/components/Checkbox'

describe('Checkbox', () => {
  it('renders default checkbox', () => {
    const wrapper = shallowMount(Checkbox, {
      propsData: {
        checkboxValue: 10
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('check event value', () => {
    const wrapper = mount(Checkbox, {
      propsData: {
        checkboxValue: 10
      }
    })
    const input = wrapper.find('input')
    input.trigger('click')
    expect(wrapper.emitted('input')).toBeTruthy()
    expect(wrapper.emitted('input')[0][0]).toEqual(10)

    wrapper.setProps({ value: false })
    expect(wrapper.emitted('input')).toBeTruthy()
  })
})
