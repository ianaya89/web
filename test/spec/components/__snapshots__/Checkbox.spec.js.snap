// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders default checkbox 1`] = `
<v-checkbox-stub
  backgroundcolor=""
  checkboxvalue="10"
  class="checkbox"
  color="red"
  errorcount="1"
  errormessages=""
  indeterminateicon="$vuetify.icons.checkboxIndeterminate"
  label="Label"
  messages=""
  officon="$vuetify.icons.checkboxOff"
  onicon="$vuetify.icons.checkboxOn"
  ripple="true"
  rules=""
  successmessages=""
  value="10"
  valuecomparator="function deepEqual(a, b) {
    if (a === b) return true;
    if (a instanceof Date && b instanceof Date) {
        // If the values are Date, they were convert to timestamp with getTime and compare it
        if (a.getTime() !== b.getTime()) return false;
    }
    if (a !== Object(a) || b !== Object(b)) {
        // If the values aren't objects, they were already checked for equality
        return false;
    }
    var props = Object.keys(a);
    if (props.length !== Object.keys(b).length) {
        // Different number of props, don't bother to check
        return false;
    }
    return props.every(function (p) {
        return deepEqual(a[p], b[p]);
    });
}"
/>
`;
